### PR TITLE
Throw an exception with msg rather than error() with it.

### PR DESCRIPTION
**Pull request by petershintech**
Try to throw a similar message with other BoundsErrors but not sure the given message is the best one for this case. Please feel free to revise the message. #1868

### CODE REVIEW

File: src/other/index.jl [hunk 1]
```diff
@@ -128,7 +128,7 @@ function Base.empty!(x::Index)
 end
 
 function Base.insert!(x::Index, idx::Integer, nm::Symbol)
-    1 <= idx <= length(x.names)+1 || error(BoundsError())
+    1 <= idx <= length(x.names)+1 || throw(BoundsError("attempt to insert to a data frame with $(length(x.names)) columns at index $idx"))
```
Comment (bkamins) [inline src/other/index.jl]:
```suggestion
    if !(1 <= idx <= length(x.names)+1)
    throw(BoundsError("attempt to insert to a data frame with $(length(x.names)) columns at index $idx"))
     end
```

Comment (bkamins) [reply src/other/index.jl]: Looks good to me, but @nalimilan has a better feel for language issues 😄. Thank you.

Comment (nalimilan) [reply src/other/index.jl]: Maybe this?
```suggestion
    if !(1 <= idx <= length(x.names)+1)
        throw(BoundsError("attempt to insert a column to a data frame with $(length(x.names)) columns at index $idx"))
     end
```

**Comment by petershintech** [comment]
Agree. I committed @nalimilan's code to the branch. Would you merge the branch?

**Comment by petershintech** [comment]
In test/dataframe.jl, I can't see any unit test checking insert!(). Where should I insert a unit test for this new code? Any suggestion?

**Comment by bkamins** [comment]
Thank you

**Comment by bkamins** [comment]
Adding test can be a separate PR. You can add test of `Index` `insert!` in test/index.jl.

**Pull request merged**